samples/csharp/getting-started/Ranking_Web/README.md

```diff
@@ -0,0 +1,199 @@
+
+Ranking is useful to any scenario where it is important to list items in an order that increases the likelihood of a click, purchase, reservation, etc.
+ 
+In this sample, we show how to apply ranking to search engine results.  To perform ranking, there are two algorithms currently available - FastTree Boosting (FastRank) and Light Gradient Boosting Machine (LightGBM).  We use the LightGBM's LambdaRank implementation in this sample to automatically build an ML model to predict ranking. 
+
+## Dataset
+The data used by this sample is based on a public [dataset provided by Microsoft](https://www.microsoft.com/en-us/research/project/mslr/) originally provided Microsoft Bing. The dataset is released under a [CC-by 4.0](https://creativecommons.org/licenses/by/4.0/) license and includes training, validation, and testing data.
+
+```
+@article{DBLP:journals/corr/QinL13,
+  author    = {Tao Qin and 
+               Tie{-}Yan Liu},
+  title     = {Introducing {LETOR} 4.0 Datasets},
+  journal   = {CoRR},
+  volume    = {abs/1306.2597},
+  year      = {2013},
+  url       = {https://arxiv.org/abs/1306.2597},
+  timestamp = {Mon, 01 Jul 2013 20:31:25 +0200},
+  biburl    = {https://dblp.uni-trier.de/rec/bib/journals/corr/QinL13},
+  bibsource = {dblp computer science bibliography, https://dblp.org}
+}
+```
+
+The following description is provided for this dataset:
+
+The datasets are machine learning data, in which queries and urls are represented by IDs. The datasets consist of feature vectors extracted from query-url pairs along with relevance judgment labels:
+
+* The relevance judgments are obtained from a retired labeling set of a commercial web search engine (Microsoft Bing), which take 5 values from 0 (irrelevant) to 4 (perfectly relevant).
+
+* The features are basically extracted by us (e.g. Microsoft), and are those widely used in the research community.
+
+In the data files, each row corresponds to a query-url pair. The first column is relevance label of the pair, the second column is query id, and the following columns are features. The larger value the relevance label has, the more relevant the query-url pair is. A query-url pair is represented by a 136-dimensional feature vector.
+
+## ML Task - Ranking
+As previously mentioned, this sample uses the LightGBM LambdaRank algorithm which is applied using a supervised learning technique known as [**Learning to Rank**](https://en.wikipedia.org/wiki/Learning_to_rank).  This technique requires that train/validation/test datasets contain groups of data instances that are each labeled with their relevance score (e.g. relevance judgment label).  The label is a numerical/ordinal value, such as {0, 1, 2, 3, 4}.  The process for labeling these data instances with their relevance scores can be done manually by subject matter experts.  Or, the labels can be determined using other metrics, such as the number of clicks on a given search result. 
```

Comment:
May want to a search/replace for "`.<space><space>`" with "`.<space>`". I think our docs generally have one space after a sentence. This markdown file has a mix of single & double.

```suggestion
As previously mentioned, this sample uses the LightGBM LambdaRank algorithm which is applied using a supervised learning technique known as [**Learning to Rank**](https://en.wikipedia.org/wiki/Learning_to_rank). This technique requires that train/validation/test datasets contain groups of data instances that are each labeled with their relevance score (e.g. relevance judgment label). The label is a numerical/ordinal value, such as {0, 1, 2, 3, 4}. The process for labeling these data instances with their relevance scores can be done manually by subject matter experts. Or, the labels can be determined using other metrics, such as the number of clicks on a given search result. 
```

Reply:
#resolved